test-dist/testnew.go

```diff
@@ -279,11 +285,19 @@ func versionMatch(a, b string) bool {
 
 func testFileAdd(tdir, bin string) error {
 	stump.VLog("  - checking that we can add and cat a file")
-	text := "hello world! This node should work"
-	data := bytes.NewBufferString(text)
-	c := exec.Command(bin, "add", "-q", "--progress=false")
-	c.Env = []string{"IPFS_PATH=" + tdir}
-	c.Stdin = data
+	text := []byte("hello world! This node should work")
+	testFile := filepath.Join(tdir, "/test.txt")
+	err := ioutil.WriteFile(testFile, text, 0644)
+	if err != nil {
+		stump.Error("testfileadd could not create test file: %s", err)
+	}
+
+	c := exec.Command(bin, "add", "-q", "--progress=false", testFile)
+	if runtime.GOOS == "windows" {
+		c.Env = os.Environ()
+	}
+	c.Env = append(c.Env, "IPFS_PATH="+tdir)
```

Comment:
You have to unfortunately check for duplicate environment variables as duplicates are not defined until Go 1.9. This might break someone's IPFS_PATH if it is explicitly set.